prov/shm/src/smr_init.c

```diff
@@ -66,8 +66,12 @@ static int smr_getinfo(uint32_t version, const char *node, const char *service,
 		       struct fi_info **info)
 {
 	struct fi_info *cur;
+	int fast_rma;
 	int ret;
 
+	fast_rma = hints ? smr_fast_rma_enabled(hints->domain_attr->mr_mode,
+						hints->tx_attr->msg_order) : 0;
```

Comment:
merging, but please look at this test.  There's the potential to de-reference a null pointer.  Also, I think you can enable fast_rma if no hints are provided.  I think the correct check sequence would be:
```
mr_mode = hints && hints->domain_attr ? hints->domain-attr : FI_MR_VIRT_ADDR;
msg_order = hints && hints->tx_attr ? hints->tx_attr->msg_order : FI_ORDER_SAS;

fast_rma = smr_fast_rma_enabled(mr_mode, msg_order);
```
see below